docs/develop/go/test-suites.mdx

```diff
@@ -160,6 +161,103 @@ When an Activity sends a Heartbeat, be sure that you can see the Heartbeats in y
 
 If an Activity is supposed to react to a Cancellation, you can test whether it reacts correctly by canceling it.
 
+## Mock and override Nexus operations
+
+A workflow that executes a Nexus operation can be tested by providing a mock of the Nexus operation
+or overriding its implementation with the Nexus mock environment. This allows you to test the
+workflow without needing to create a Nexus handler service to run the Nexus operation.
+
+Let's take a look at a test that simulates a Nexus operation call. Assume the Nexus operation is
+called `sample-operation`, the input type is `SampleInput`, the output type is `SampleOutput`, and
+it belongs to the Nexus service `sample-service`.
+
+```go
+func (s *UnitTestSuite) Test_SimpleWorkflow_NexusSyncOperation() {
+        s.env.OnNexusOperation(
+                "sample-service",
+                nexus.NewOperationReference[SampleInput, SampleOutput]("sample-operation"),
+                SampleInput{},
+                workflow.NexusOperationOptions{},
+        ).Return(
+                &nexus.HandlerStartOperationResultSync[SampleOutput]{
+                        Value: SampleOutput{},
+                },
+                nil, // error if you want to simulate an error in the ExecuteOperation call
+        )
+        // You can also add a delay to return the mock values by calling After().
+        // Eg: s.env.OnNexusOperation(...).Return(...).After(1*time.Second)
+
+        s.env.ExecuteWorkflow(SimpleWorkflow, "test_nexus_operation")
+
+        s.True(s.env.IsWorkflowCompleted())
+        s.NoError(s.env.GetWorkflowError())
+}
+```
+
+The example above simulates a call to a Nexus synchronous operation, indicated by the returned value
+type `*nexus.HandlerStartOperationResultSync[T]`. Since `OnNexusOperation` needs to know the
```

Comment:
I found this a little confusing and I think re-ordering some of the sentences might help. 

I think (but may be wrong):

* `nexus.NewOperationReference` creates an operation reference.
* The reference includes the signature of the operation method but does not have access to implementation details.
* If you know the operation directly, you can call it? 

Why would you use the operation directly instead of the reference? This is what confused me the most. When would you have it "available"? Is it because you created an instance of a type or are using a class method? What role does this have in testing?

Reply:
Updated the text, hopefully it's more clear now.